core/math/math_funcs.h

```diff
@@ -447,14 +447,22 @@ class Math {
 
 	static _ALWAYS_INLINE_ double smoothstep(double p_from, double p_to, double p_s) {
 		if (is_equal_approx(p_from, p_to)) {
-			return p_from;
+			if (p_from <= p_to) {
```

Comment:
The more I think of this, the more I wonder if this check makes any sense `p_from` is approximately equal to `p_to`, they could flip order just by a floating point precision error. I think in the case of then being approximately equal we should just do `return p_s <= p_from ? 0.0 : 1.0;`

Reply:
Consider using `likely()` and `unlikely()` hints to guide the compiler towards doing better optimization.